src/Data/Codec/JSON/Record.purs

```diff
@@ -53,39 +82,80 @@ optional = Optional
 -- | The class used to enable the building of `Record` codecs by providing a
 -- | record of codecs.
 class RowListCodec (rl ∷ RL.RowList Type) (ri ∷ Row Type) (ro ∷ Row Type) | rl → ri ro where
-  rowListCodec ∷ ∀ proxy. proxy rl → Record ri → CJ.PropCodec (Record ro)
+  rowListCodec ∷ Record ri → CJ.PropCodec (Record ro)
 
-instance rowListCodecNil ∷ RowListCodec RL.Nil () () where
-  rowListCodec _ _ = CJ.record
+instance RowListCodec RL.Nil () () where
+  rowListCodec _ = CJ.record
 
-instance rowListCodecConsOptional ∷
+instance
   ( RowListCodec rs ri' ro'
   , R.Cons sym (Optional a) ri' ri
   , R.Cons sym (Maybe a) ro' ro
   , IsSymbol sym
   ) ⇒
   RowListCodec (RL.Cons sym (Optional a) rs) ri ro where
-  rowListCodec _ codecs =
+  rowListCodec codecs =
     CJ.recordPropOptional (Proxy ∷ Proxy sym) codec tail
     where
     codec ∷ CJ.Codec a
     codec = coerce (Rec.get (Proxy ∷ Proxy sym) codecs ∷ Optional a)
 
     tail ∷ CJ.PropCodec (Record ro')
-    tail = rowListCodec (Proxy ∷ Proxy rs) ((unsafeCoerce ∷ Record ri → Record ri') codecs)
+    tail = rowListCodec @rs ((unsafeCoerce ∷ Record ri → Record ri') codecs)
 
-else instance rowListCodecCons ∷
+else instance
   ( RowListCodec rs ri' ro'
   , R.Cons sym (CJ.Codec a) ri' ri
   , R.Cons sym a ro' ro
   , IsSymbol sym
   ) ⇒
   RowListCodec (RL.Cons sym (CJ.Codec a) rs) ri ro where
-  rowListCodec _ codecs =
+  rowListCodec codecs =
     CJ.recordProp (Proxy ∷ Proxy sym) codec tail
     where
     codec ∷ CJ.Codec a
     codec = Rec.get (Proxy ∷ Proxy sym) codecs
 
     tail ∷ CJ.PropCodec (Record ro')
-    tail = rowListCodec (Proxy ∷ Proxy rs) ((unsafeCoerce ∷ Record ri → Record ri') codecs)
+    tail = rowListCodec @rs ((unsafeCoerce ∷ Record ri → Record ri') codecs)
+
+
+-- | The class used to enable the building of `Record` codecs by providing a
+-- | record of codecs.
+class RowListCodecStrict (rl ∷ RL.RowList Type) (ri ∷ Row Type) (ro ∷ Row Type) | rl → ri ro where
```

Comment:
I looked at parametrizing this class or its method, but that turned out to be almost as much ceremony as duplicating it. And since we're duplicating everything else anyway...